app/utils/communication/notifications.py

```diff
@@ -292,51 +242,35 @@ async def send_notification_to_user(
                 f"Notification: Unable to send firebase notification to user {user_id} with device: {error}",
             )
 
+    async def send_notification_to_users(
+        self,
+        user_ids: list[str],
+        message: Message,
+        db: AsyncSession,
+    ) -> None:
+        for user_id in user_ids:
+            await self.send_notification_to_user(
+                user_id=user_id,
+                message=message,
+                db=db,
+            )
```

Comment:
Here you will make one firebase request per user. You may prefer to get all firebase tokens for your users and then call:
```python
await self._send_firebase_trigger_notification_by_tokens(
                tokens=firebase_device_tokens,
                db=db,
            )
```
---
`send_notification_to_user(user_id: str)` could call `send_notification_to_users(user_ids=[user_id])`

Reply:
I agree with the first part but I don't really understand why send_notification_to_user(user_id: str) should call send_notification_to_users(user_ids=[user_id]) ?

Reply:
I wouldn't say it should, but it may be a way to reduce code duplication between `send_notification_to_user` and `send_notification_to_users`

Reply:
I don't see any code duplication (if you meant in `notification.py`

Reply:
I don't see any code duplication (if you meant in `notification.py`)

Reply:
I don't see any code duplication (if you meant in `notification.py`)